radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStateComputer.java

```diff
@@ -63,20 +64,20 @@
 
 	private final Serialization serialization;
 	private final RadixEngine<LedgerAtom> radixEngine;
-	private final View epochChangeView;
+	private final View epochCeilingView;
 
 	public RadixEngineStateComputer(
 		Serialization serialization,
 		RadixEngine<LedgerAtom> radixEngine,
-		View epochChangeView
+		View epochCeilingView
 	) {
-		if (epochChangeView.isGenesis()) {
+		if (epochCeilingView.isGenesis()) {
```

Comment:
Perhaps it better to use factory method and check preconditions there instead of throwing exception in constructor.

Reply:
What's the reasoning for doing it in that way?